src/Microsoft.DotNet.CodeFormatting/Analyzers/OptimizeNamespaceImportsFixer.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.DotNet.CodeFormatting.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class OptimizeNamespaceImportsFixer : CodeFixProvider
+    {
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            Diagnostic diagnostic = context.Diagnostics.First();
+            var usingDirectiveNodes = new List<SyntaxNode>();
+
+            // We recapitulate the primary diagnostic location in the 
```

Comment:
OK, this addresses my comment above as far as why you did it. Sri says that it was intended just to have the "others", but that is not documented, and Roslyn itself doesn't rely on that behavior anywhere. So I guess this is fine.